In [1]:
import pandas as pd
import sqlite3



In [2]:
conn = sqlite3.connect("belly_button_biodiversity.sqlite")


In [3]:
otu_df = pd.read_sql_query("select * from otu limit 5;", conn)
otu_df

otu_id                        lowest_taxonomic_unit_found
0       1  Archaea;Euryarchaeota;Halobacteria;Halobacteri...
1       2  Archaea;Euryarchaeota;Halobacteria;Halobacteri...
2       3  Archaea;Euryarchaeota;Halobacteria;Halobacteri...
3       4  Archaea;Euryarchaeota;Methanobacteria;Methanob...
4       5  Archaea;Euryarchaeota;Methanobacteria;Methanob...

In [28]:
otu_df = pd.read_sql_query("select * from otu;", conn)


In [29]:
otu_df

otu_id                        lowest_taxonomic_unit_found
0          1  Archaea;Euryarchaeota;Halobacteria;Halobacteri...
1          2  Archaea;Euryarchaeota;Halobacteria;Halobacteri...
2          3  Archaea;Euryarchaeota;Halobacteria;Halobacteri...
3          4  Archaea;Euryarchaeota;Methanobacteria;Methanob...
4          5  Archaea;Euryarchaeota;Methanobacteria;Methanob...
5          6  Archaea;Euryarchaeota;Methanobacteria;Methanob...
6          7                                           Bacteria
7          8                                           Bacteria
8          9                                           Bacteria
9         10                                           Bacteria
10        11                                           Bacteria
11        12                                           Bacteria
12        13                                           Bacteria
13        14                                           Bacteria
14        15                                           Bacteria
15        16                                           Bacteria
16        17                                           Bacteria
17        18                                           Bacteria
18        19                                           Bacteria
19        20                                           Bacteria
20        21                                           Bacteria
21        22                                           Bacteria
22        23                                           Bacteria
23        24                                           Bacteria
24        25                                           Bacteria
25        26                                           Bacteria
26        27                                           Bacteria
27        28                                           Bacteria
28        29                                           Bacteria
29        30                                           Bacteria
...      ...                                                ...
3644    3645  Bacteria;Synergistetes;Synergistia;Synergistal...
3645    3646  Bacteria;Synergistetes;Synergistia;Synergistal...
3646    3647  Bacteria;Synergistetes;Synergistia;Synergistal...
3647    3648  Bacteria;Synergistetes;Synergistia;Synergistal...
3648    3649  Bacteria;Synergistetes;Synergistia;Synergistal...
3649    3650  Bacteria;Synergistetes;Synergistia;Synergistal...
3650    3651  Bacteria;Synergistetes;Synergistia;Synergistal...
3651    3652  Bacteria;Synergistetes;Synergistia;Synergistal...
3652    3653  Bacteria;Synergistetes;Synergistia;Synergistal...
3653    3654  Bacteria;Synergistetes;Synergistia;Synergistal...
3654    3655  Bacteria;Synergistetes;Synergistia;Synergistal...
3655    3656  Bacteria;Synergistetes;Synergistia;Synergistal...
3656    3657  Bacteria;Synergistetes;Synergistia;Synergistal...
3657    3658  Bacteria;Synergistetes;Synergistia;Synergistal...
3658    3659  Bacteria;Synergistetes;Synergistia;Synergistal...
3659    3660  Bacteria;Synergistetes;Synergistia;Synergistal...
3660    3661  Bacteria;Synergistetes;Synergistia;Synergistal...
3661    3662             Bacteria;TM7;TM7_genera_incertae_sedis
3662    3663  Bacteria;Verrucomicrobia;Spartobacteria;Sparto...
3663    3664  Bacteria;Verrucomicrobia;Spartobacteria;Sparto...
3664    3665  Bacteria;Verrucomicrobia;Spartobacteria;Sparto...
3665    3666  Bacteria;Verrucomicrobia;Verrucomicrobiae;Verr...
3666    3667                                       Unclassified
3667    3668                                       Unclassified
3668    3669                                       Unclassified
3669    3670                                       Unclassified
3670    3671                                       Unclassified
3671    3672                                       Unclassified
3672    3673                                       Unclassified
3673    3674                                       Unclassified

[3674 rows x 2 columns]

In [5]:
samples_df = pd.read_sql_query("select * from samples;", conn)
samples_df

otu_id  BB_940  BB_941  BB_943  BB_944  BB_945  BB_946  BB_947  BB_948  \
0          1       0       0       0       0       0       0       0       0   
1          2       0       0       0       0       0       0       0       0   
2          3       0       0       0       0       0       0       0       0   
3          4       0       0       0       0       0       0       0       0   
4          5       0       0       0       0       0       0       0       0   
5          6       0       0       0       0       0       0       0       0   
6          7       0       0       0       0       0       0       0       0   
7          8       0       0       0       0       0       0       0       0   
8          9       0       0       0       0       0       1       0       0   
9         10       0       0       0       0       0       0       0       0   
10        11       1       0       0       0       0       0       0       1   
11        12       0       0       0       0       0       0       0       0   
12        13       0       0       0       0       0       0       0       0   
13        14       0       0       0       0       0       0       0       0   
14        15       0       0       0       0       0       0       0       0   
15        16       0       0       0       0       0       0       0       0   
16        17       0       0       0       0       0       0       0       0   
17        18       0       0       0       0       0       2       1       0   
18        19       1       0       0       0       0       0       0       0   
19        20       0       0       0       0       0       0       0       0   
20        21       0       0       0       0       0       0       0       0   
21        22       1       0       0       0       0       0       0       0   
22        23       0       0       0       0       0       0       0       0   
23        24       1       0       0       0       0       0       0       0   
24        25       0       0       0       0       0       0       0       0   
25        26       0       0       0       0       0       0       0       0   
26        27       0       0       0       0       0       0       0       0   
27        28       0       0       0       0       0       0       0       0   
28        29       0       0       0       0       0       0       0       0   
29        30       0       0       0       0       0       0       0       0   
...      ...     ...     ...     ...     ...     ...     ...     ...     ...   
3644    3645       0       0       0       0       0       0       0       0   
3645    3646       0       0       0       0       0       0       0       0   
3646    3647       0       0       0       0       0       0       0       0   
3647    3648       0       0       0       0       0       0       0       0   
3648    3649       0       0       0       0       0       0       0       0   
3649    3650       0       0       0       0       0       0       0       0   
3650    3651       0       0       0       0       0       0       0       0   
3651    3652       0       0       0       0       0       0       0       0   
3652    3653       0       0       0       0       0       0       0       0   
3653    3654       0       0       0       0       0       0       0       0   
3654    3655       0       0       0       0       0       0       0       0   
3655    3656       0       0       0       0       0       0       0       0   
3656    3657       0       0       0       0       0       0       0       0   
3657    3658       0       0       0       0       0       0       0       0   
3658    3659       0       0       0       0       0       0       0       0   
3659    3660       0       0       0       0       0       0       0       0   
3660    3661       0       0       0       0       0       0       0       0   
3661    3662       0       0       0       0       0       0       0       0   
3662  

In [6]:
sampMeta_df = pd.read_sql_query("select * from samples_metadata;", conn)
sampMeta_df

SAMPLEID                      EVENT              ETHNICITY GENDER  AGE  \
0         940  BellyButtonsScienceOnline              Caucasian      F   24   
1         941                          0  Caucasian/Midleastern      F   34   
2         943  BellyButtonsScienceOnline              Caucasian      F   49   
3         944  BellyButtonsScienceOnline               European      M   44   
4         945  BellyButtonsScienceOnline              Caucasian      F   48   
5         946  BellyButtonsScienceOnline              Caucasian      F   42   
6         947  BellyButtonsScienceOnline              Caucasian      M   49   
7         948  BellyButtonsScienceOnline              Caucasian      M   20   
8         949  BellyButtonsScienceOnline              Caucasian      F   51   
9         950  BellyButtonsScienceOnline              Caucasian      M   51   
10        952  BellyButtonsScienceOnline              Caucasian      F   51   
11        953  BellyButtonsScienceOnline              Caucasian      F   38   
12        954  BellyButtonsScienceOnline              Caucasian      M   26   
13        955  BellyButtonsScienceOnline              Caucasian      F   27   
14        956  BellyButtonsScienceOnline              Caucasian      M   16   
15        958  BellyButtonsScienceOnline              Caucasian      M   43   
16        959  BellyButtonsScienceOnline              Caucasian      M   40   
17        960  BellyButtonsScienceOnline              Caucasian      M   48   
18        961  BellyButtonsScienceOnline              Caucasian      F   35   
19        962  BellyButtonsScienceOnline              Caucasian      F   56   
20        963  BellyButtonsScienceOnline              Caucasian      F   33   
21        964  BellyButtonsScienceOnline              Caucasian      F   43   
22        966  BellyButtonsScienceOnline              Caucasian      M   56   
23        967  BellyButtonsScienceOnline       Caucasian/Jewish      F   43   
24        968  BellyButtonsScienceOnline              Caucasian      F   38   
25        969  BellyButtonsScienceOnline        Caucasian/Asian      F   31   
26        970  BellyButtonsScienceOnline              Caucasian      F   42   
27        971  BellyButtonsScienceOnline           Asian(South)      M   36   
28        972  BellyButtonsScienceOnline              Caucasian      F   31   
29        973  BellyButtonsScienceOnline              Caucasian      F    4   
..        ...                        ...                    ...    ...  ...   
123      1540      BellyButtonsWildLives              Caucasian      M   30   
124      1541      BellyButtonsWildLives              Caucasian      F   26   
125      1542      BellyButtonsWildLives              Caucasian      M   36   
126      1543      BellyButtonsWildLives              Caucasian      F   34   
127      1544      BellyButtonsWildLives              Caucasian      F   33   
128      1545      BellyButtonsWildLives              Caucasian      F   39   
129      1546      BellyButtonsWildLives              Caucasian      M   34   
130      1547      BellyButtonsWildLives                      0      M   38   
131      1548      BellyButtonsWildLives              Caucasian      M    6   
132      1549      BellyButtonsWildLives              Caucasian      F   37   
133      1550      BellyButtonsWildLives              Caucasian      F   29   
134      1551      BellyButtonsWildLives              Caucasian      M   26   
135      1552      BellyButtonsWildLives              Caucasian      M   25   
136      1553      BellyButtonsWildLives              Caucasian      F   50   
137      1554      BellyButtonsWildLives              Caucasian      F   18   
138      1555      BellyButtonsWildLives              Caucasian      M   26   
139      1556      BellyButtonsWildLives               Hispanic      M   34   
140      1557      BellyButtonsWildLives              Caucasian      F   26   
141      1558      BellyButtonsWildLives              C

In [7]:
samples_df.columns

Index(['otu_id', 'BB_940', 'BB_941', 'BB_943', 'BB_944', 'BB_945', 'BB_946',
       'BB_947', 'BB_948', 'BB_949',
       ...
       'BB_1562', 'BB_1563', 'BB_1564', 'BB_1572', 'BB_1573', 'BB_1574',
       'BB_1576', 'BB_1577', 'BB_1581', 'BB_1601'],
      dtype='object', length=154)

In [8]:
samples_list = samples_df.columns

In [9]:
samples_list

Index(['otu_id', 'BB_940', 'BB_941', 'BB_943', 'BB_944', 'BB_945', 'BB_946',
       'BB_947', 'BB_948', 'BB_949',
       ...
       'BB_1562', 'BB_1563', 'BB_1564', 'BB_1572', 'BB_1573', 'BB_1574',
       'BB_1576', 'BB_1577', 'BB_1581', 'BB_1601'],
      dtype='object', length=154)

In [22]:
type(samples_list)

pandas.core.indexes.base.Index

In [57]:
x = len(samples_list)

In [58]:
x

154

In [62]:
cnt = 0
cleanList = []

In [69]:

for i in range(1,x):
    cleanList.append(samples_list[i])
  

In [84]:
otuDesc =[]

In [ ]:
for i in range(1,x):
    cleanList.append(samples_list[i])

In [73]:
otu_df.head()

otu_id                        lowest_taxonomic_unit_found
0       1  Archaea;Euryarchaeota;Halobacteria;Halobacteri...
1       2  Archaea;Euryarchaeota;Halobacteria;Halobacteri...
2       3  Archaea;Euryarchaeota;Halobacteria;Halobacteri...
3       4  Archaea;Euryarchaeota;Methanobacteria;Methanob...
4       5  Archaea;Euryarchaeota;Methanobacteria;Methanob...

In [81]:
otu_df.iloc[6,1]

'Bacteria'

In [83]:
len(otu_df["otu_id"])

3674

In [85]:
for index in range(len(otu_df["otu_id"])):
    otuDesc.append(otu_df.iloc[index,1])

In [86]:
otuDesc

['Archaea;Euryarchaeota;Halobacteria;Halobacteriales;Halobacteriaceae;Halococcus',
 'Archaea;Euryarchaeota;Halobacteria;Halobacteriales;Halobacteriaceae;Halococcus',
 'Archaea;Euryarchaeota;Halobacteria;Halobacteriales;Halobacteriaceae;Natronorubrum',
 'Archaea;Euryarchaeota;Methanobacteria;Methanobacteriales;Methanobacteriaceae;Methanobrevibacter',
 'Archaea;Euryarchaeota;Methanobacteria;Methanobacteriales;Methanobacteriaceae;Methanobrevibacter',
 'Archaea;Euryarchaeota;Methanobacteria;Methanobacteriales;Methanobacteriaceae;Methanobrevibacter',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bacteria',
 'Bact

    Args: Sample in the format: `BB_940`

    Returns a json dictionary of sample metadata in the format

    {
        AGE: 24,
        BBTYPE: "I",
        ETHNICITY: "Caucasian",
        GENDER: "F",
        LOCATION: "Beaufort/NC",
        SAMPLEID: 940
    }


In [11]:
sample = "BB_940"

In [12]:
data = sample.split("_")

In [13]:
data[0]

'BB'

In [14]:
idBB = data[1]

In [15]:
type(idBB)

str

In [16]:
sampMeta_df['SAMPLEID']

0       940
1       941
2       943
3       944
4       945
5       946
6       947
7       948
8       949
9       950
10      952
11      953
12      954
13      955
14      956
15      958
16      959
17      960
18      961
19      962
20      963
21      964
22      966
23      967
24      968
25      969
26      970
27      971
28      972
29      973
       ... 
123    1540
124    1541
125    1542
126    1543
127    1544
128    1545
129    1546
130    1547
131    1548
132    1549
133    1550
134    1551
135    1552
136    1553
137    1554
138    1555
139    1556
140    1557
141    1558
142    1561
143    1562
144    1563
145    1564
146    1572
147    1573
148    1574
149    1576
150    1577
151    1581
152    1601
Name: SAMPLEID, Length: 153, dtype: int64

In [17]:
len(sampMeta_df['SAMPLEID'])

153

In [18]:
bb_df = sampMeta_df.groupby(['SAMPLEID'] == idBB)

KeyError: False

In [19]:
bb_df = sampMeta_df.loc[sampMeta_df['SAMPLEID'] == str(idBB),:]

# clean_ufo_df.loc[clean_ufo_df["country"] == "us",:]

/Users/scott/anaconda3/lib/python3.6/site-packages/pandas/core/ops.py:816: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  result = getattr(x, name)(y)


TypeError: invalid type comparison

In [20]:
type(sampMeta_df)

pandas.core.frame.DataFrame

In [21]:
sampMeta_df.iloc[sampMeta_df[0,1]

SyntaxError: unexpected EOF while parsing (<ipython-input-21-b9b2d3f2db8f>, line 1)

In [22]:
idResult_df = sampMeta_df.loc[sampMeta_df['SAMPLEID'] == int(idBB)]

In [23]:
it = idResult_df.iloc[0]['SAMPLEID']

In [24]:
print(it)

940


In [25]:
results = [{
        'AGE': int(idResult_df.iloc[0]['AGE']),
        "BBTYPE": idResult_df.iloc[0]['BBTYPE'],
        "ETHNICITY": idResult_df.iloc[0]['ETHNICITY'],
        "GENDER": idResult_df.iloc[0]['GENDER'],
        "LOCATION": idResult_df.iloc[0]['LOCATION'],
        "SAMPLEID": idResult_df.iloc[0]['SAMPLEID']
    }]

In [245]:
x = int(idResult_df.iloc[0]['AGE'])

In [26]:
    print(sample)
    
    # sample = "BB_940" 
    data = sample.split("_")
    idBB = data[1] 
    print(sampMeta_df.loc[sampMeta_df['SAMPLEID'] == int(idBB)])  
    idResult_df = sampMeta_df.loc[sampMeta_df['SAMPLEID'] == int(idBB)]
    results = [{
        'AGE': int(idResult_df.iloc[0]['AGE']),
        "BBTYPE": idResult_df.iloc[0]['BBTYPE'],
        "ETHNICITY": idResult_df.iloc[0]['ETHNICITY'],
        "GENDER": idResult_df.iloc[0]['GENDER'],
        "LOCATION": idResult_df.iloc[0]['LOCATION'],
        "SAMPLEID": idResult_df.iloc[0]['SAMPLEID']
    }]

#     return jsonify(results)

BB_940
   SAMPLEID                      EVENT  ETHNICITY GENDER  AGE  WFREQ BBTYPE  \
0       940  BellyButtonsScienceOnline  Caucasian      F   24    2.0      I   

      LOCATION COUNTRY012 ZIP012   ...    DOG CAT IMPSURFACE013     NPP013  \
0  Beaufort/NC        usa  22306   ...     no  no          8852  37.172222   

   MMAXTEMP013  PFC013  IMPSURFACE1319  NPP1319 MMAXTEMP1319  PFC1319  
0         54.5     0.0               1      0.0    33.990002     25.5  

[1 rows x 22 columns]


In [246]:
type(x)

int

In [243]:
xx


24

In [244]:
type(xx)

int

In [235]:
print(results)


[{'AGE': 24, 'BBTYPE': 'I', 'ETHNICITY': 'Caucasian', 'GENDER': 'F', 'LOCATION': 'Beaufort/NC', 'SAMPLEID': 940}]


In [144]:
sample = "BB_940"
data = sample.split("_")
wfreqID = data[1] 

In [145]:
wfreqResult_df = sampMeta_df.loc[sampMeta_df['SAMPLEID'] == int(idBB)]

In [146]:
wfreqResult_df

SAMPLEID                      EVENT  ETHNICITY GENDER  AGE  WFREQ BBTYPE  \
0       940  BellyButtonsScienceOnline  Caucasian      F   24    2.0      I   

      LOCATION COUNTRY012 ZIP012   ...    DOG CAT IMPSURFACE013     NPP013  \
0  Beaufort/NC        usa  22306   ...     no  no          8852  37.172222   

   MMAXTEMP013  PFC013  IMPSURFACE1319  NPP1319 MMAXTEMP1319  PFC1319  
0         54.5     0.0               1      0.0    33.990002     25.5  

[1 rows x 22 columns]

In [147]:
wfreqNbr = wfreqResult_df.iloc[0]['WFREQ']

In [148]:
wfreqNbr

2.0

In [150]:
otu_df.head()


otu_id                        lowest_taxonomic_unit_found
0       1  Archaea;Euryarchaeota;Halobacteria;Halobacteri...
1       2  Archaea;Euryarchaeota;Halobacteria;Halobacteri...
2       3  Archaea;Euryarchaeota;Halobacteria;Halobacteri...
3       4  Archaea;Euryarchaeota;Methanobacteria;Methanob...
4       5  Archaea;Euryarchaeota;Methanobacteria;Methanob...

In [164]:
samples_df.head()

otu_id  BB_940  BB_941  BB_943  BB_944  BB_945  BB_946  BB_947  BB_948  \
0       1       0       0       0       0       0       0       0       0   
1       2       0       0       0       0       0       0       0       0   
2       3       0       0       0       0       0       0       0       0   
3       4       0       0       0       0       0       0       0       0   
4       5       0       0       0       0       0       0       0       0   

   BB_949   ...     BB_1562  BB_1563  BB_1564  BB_1572  BB_1573  BB_1574  \
0       0   ...           0        0        0        0        0        0   
1       0   ...           0        0        0        0        0        0   
2       0   ...           0        0        0        0        0        0   
3       0   ...           0        0        0        0        0        0   
4       0   ...           0        0        0        0        0        0   

   BB_1576  BB_1577  BB_1581  BB_1601  
0        0        0        0        0  
1        0        0        0        0  
2        0        0        0        0  
3        0        0        0        0  
4        0        0        0        0  

[5 rows x 154 columns]

In [165]:
samples_df

otu_id  BB_940  BB_941  BB_943  BB_944  BB_945  BB_946  BB_947  BB_948  \
0          1       0       0       0       0       0       0       0       0   
1          2       0       0       0       0       0       0       0       0   
2          3       0       0       0       0       0       0       0       0   
3          4       0       0       0       0       0       0       0       0   
4          5       0       0       0       0       0       0       0       0   
5          6       0       0       0       0       0       0       0       0   
6          7       0       0       0       0       0       0       0       0   
7          8       0       0       0       0       0       0       0       0   
8          9       0       0       0       0       0       1       0       0   
9         10       0       0       0       0       0       0       0       0   
10        11       1       0       0       0       0       0       0       1   
11        12       0       0       0       0       0       0       0       0   
12        13       0       0       0       0       0       0       0       0   
13        14       0       0       0       0       0       0       0       0   
14        15       0       0       0       0       0       0       0       0   
15        16       0       0       0       0       0       0       0       0   
16        17       0       0       0       0       0       0       0       0   
17        18       0       0       0       0       0       2       1       0   
18        19       1       0       0       0       0       0       0       0   
19        20       0       0       0       0       0       0       0       0   
20        21       0       0       0       0       0       0       0       0   
21        22       1       0       0       0       0       0       0       0   
22        23       0       0       0       0       0       0       0       0   
23        24       1       0       0       0       0       0       0       0   
24        25       0       0       0       0       0       0       0       0   
25        26       0       0       0       0       0       0       0       0   
26        27       0       0       0       0       0       0       0       0   
27        28       0       0       0       0       0       0       0       0   
28        29       0       0       0       0       0       0       0       0   
29        30       0       0       0       0       0       0       0       0   
...      ...     ...     ...     ...     ...     ...     ...     ...     ...   
3644    3645       0       0       0       0       0       0       0       0   
3645    3646       0       0       0       0       0       0       0       0   
3646    3647       0       0       0       0       0       0       0       0   
3647    3648       0       0       0       0       0       0       0       0   
3648    3649       0       0       0       0       0       0       0       0   
3649    3650       0       0       0       0       0       0       0       0   
3650    3651       0       0       0       0       0       0       0       0   
3651    3652       0       0       0       0       0       0       0       0   
3652    3653       0       0       0       0       0       0       0       0   
3653    3654       0       0       0       0       0       0       0       0   
3654    3655       0       0       0       0       0       0       0       0   
3655    3656       0       0       0       0       0       0       0       0   
3656    3657       0       0       0       0       0       0       0       0   
3657    3658       0       0       0       0       0       0       0       0   
3658    3659       0       0       0       0       0       0       0       0   
3659    3660       0       0       0       0       0       0       0       0   
3660    3661       0       0       0       0       0       0       0       0   
3661    3662       0       0       0       0       0       0       0       0   
3662  

In [174]:
descSamples_df = samples_df.sort_values('otu_id', ascending=False)

In [196]:
descSamples_df

otu_id  BB_940  BB_941  BB_943  BB_944  BB_945  BB_946  BB_947  BB_948  \
3673    3674       0       0       0       0       0       0       0       0   
3672    3673       0       0       0       0       0       0       0       0   
3671    3672       0       0       0       0       0       0       0       0   
3670    3671       0       0       0       0       0       0       0       0   
3669    3670       0       0       0       0       0       0       0       0   
3668    3669       0       0       0       0       0       0       0       0   
3667    3668       0       0       0       0       0       0       0       0   
3666    3667       0       0       0       0       0       1       0       0   
3665    3666       0       0       0       0       0       0       0       0   
3664    3665       0       0       0       0       0       0       0       0   
3663    3664       0       0       0       0       0       0       0       0   
3662    3663       0       0       0       0       0       0       0       0   
3661    3662       0       0       0       0       0       0       0       0   
3660    3661       0       0       0       0       0       0       0       0   
3659    3660       0       0       0       0       0       0       0       0   
3658    3659       0       0       0       0       0       0       0       0   
3657    3658       0       0       0       0       0       0       0       0   
3656    3657       0       0       0       0       0       0       0       0   
3655    3656       0       0       0       0       0       0       0       0   
3654    3655       0       0       0       0       0       0       0       0   
3653    3654       0       0       0       0       0       0       0       0   
3652    3653       0       0       0       0       0       0       0       0   
3651    3652       0       0       0       0       0       0       0       0   
3650    3651       0       0       0       0       0       0       0       0   
3649    3650       0       0       0       0       0       0       0       0   
3648    3649       0       0       0       0       0       0       0       0   
3647    3648       0       0       0       0       0       0       0       0   
3646    3647       0       0       0       0       0       0       0       0   
3645    3646       0       0       0       0       0       0       0       0   
3644    3645       0       0       0       0       0       0       0       0   
...      ...     ...     ...     ...     ...     ...     ...     ...     ...   
29        30       0       0       0       0       0       0       0       0   
28        29       0       0       0       0       0       0       0       0   
27        28       0       0       0       0       0       0       0       0   
26        27       0       0       0       0       0       0       0       0   
25        26       0       0       0       0       0       0       0       0   
24        25       0       0       0       0       0       0       0       0   
23        24       1       0       0       0       0       0       0       0   
22        23       0       0       0       0       0       0       0       0   
21        22       1       0       0       0       0       0       0       0   
20        21       0       0       0       0       0       0       0       0   
19        20       0       0       0       0       0       0       0       0   
18        19       1       0       0       0       0       0       0       0   
17        18       0       0       0       0       0       2       1       0   
16        17       0       0       0       0       0       0       0       0   
15        16       0       0       0       0       0       0       0       0   
14        15       0       0       0       0       0       0       0       0   
13        14       0       0       0       0       0       0       0       0   
12        13       0       0       0       0       0       0       0       0   
11    

In [199]:
df = descSamples_df.iloc[3663,1]

In [200]:
df

1

In [176]:
# descOtu.head()

    [
        {
            otu_ids: [
                1166,
                2858,
                481,
                ...
            ],
            sample_values: [
                163,
                126,
                113,
               
                ...
            ]
        }
    ]


In [228]:
sample = "BB_944"

In [229]:
position = 0
for index in range(len(clList)):
    if (clList[index] == sample):
        position = index
        break
    position += 1

In [230]:
sampleList =[]
otuList =[]
for index in range(len(descSamples_df["otu_id"])):
    if (descSamples_df.iloc[index,position] != 0):
        sampleList.append(descSamples_df.iloc[index,position])
        otuList.append(descSamples_df.iloc[index,0])

In [233]:
sampleResults = [{
        'otu_ids': otuList,
        'sample_values': sampleList
    }]

In [234]:
sampleResults

[{'otu_ids': [3555,
   3541,
   3530,
   3529,
   3527,
   3524,
   3523,
   3521,
   3517,
   3516,
   3511,
   3440,
   3368,
   3352,
   3312,
   3285,
   3282,
   3184,
   3120,
   3029,
   2945,
   2944,
   2942,
   2937,
   2722,
   2651,
   2308,
   2130,
   2106,
   2061,
   1929,
   1926,
   1925,
   1870,
   1854,
   1795,
   1770,
   1706,
   1035,
   1031,
   1013,
   1008,
   970,
   944,
   943,
   932,
   927,
   923,
   922,
   910,
   893,
   889,
   882,
   830,
   797,
   763,
   728,
   718,
   681,
   672,
   659,
   611,
   592,
   556,
   555,
   341,
   187,
   178,
   41],
  'sample_values': [33,
   4,
   1,
   1,
   3,
   2,
   2,
   2,
   1,
   5,
   1,
   2,
   7,
   1,
   3,
   1,
   1,
   1,
   4,
   1,
   1,
   1,
   1,
   2,
   2,
   3,
   4,
   2,
   1,
   1,
   3,
   1,
   1,
   1,
   1,
   11,
   5,
   1,
   2,
   1,
   18,
   1,
   2,
   8,
   19,
   1,
   1,
   5,
   278,
   1,
   1,
   1,
   1,
   1,
   1,
   1,
   2,
   1,
   1,
   1,
   1,
   1,


In [232]:
len(sampleList)

69

In [211]:
clList = descSamples_df.columns

In [213]:
len(clList)

154

In [212]:
clList[2]

'BB_941'

In [220]:
position = 0
for index in range(len(clList)):
    if (clList[index] == sample):
        position = index
        break
    position += 1
    

In [221]:
position


4

In [218]:
sample

'BB_940'